sdk/src/main/java/com/bugsnag/android/Bugsnag.java

```diff
@@ -33,8 +34,14 @@ private Bugsnag() {
      */
     @NonNull
     public static Client init(@NonNull Context androidContext) {
-        client = new Client(androidContext);
-        NativeInterface.configureClientObservers(client);
+        synchronized (lock) {
```

Comment:
why did you choose to make a `synchronized (lock) {}` block rather than label the entire static method `public static synchronized Client init()`?

Reply:
That would use the intrinsic lock on the `Bugsnag` class object, rather than a lock we've specified explicitly and made private. Even though it's a terrible idea, users could theoretically synchronise 3rd party code using the intrinsic lock on the Bugsnag class.

This approach is also more granular. We don't have to synchronise the entire contents of the method, and it's possible to use more than 1 lock in the same class, if we need to synchronise some other unrelated code at some future date.